### PR TITLE
[cxx-interop] Disable rvalue references. We don't support them correc…

### PR DESCRIPTION
**Pull request by zoecarver**
…tly (leading to lifetime issues). We are missing a consuming.

### CODE REVIEW

**Comment by zoecarver** [comment]
@swift-ci please test

**Comment by hyp** [comment]
Also what about rvalue this.

**Comment by zoecarver** [comment]
> Also what about rvalue this. I guess that can be disabled in a follow-up.

Yeah it would also be hard for a user to call an rvalue method from Swift anyway

File: lib/ClangImporter/ImportType.cpp [hunk 1]
```diff
@@ -2356,12 +2356,17 @@ ClangImporter::Implementation::importParameterType(
                  dyn_cast<clang::TemplateTypeParmType>(paramTy)) {
     swiftParamTy = findGenericTypeInGenericDecls(
         *this, templateParamType, genericParams, attrs, addImportDiagnosticFn);
-  } else if (auto refType = dyn_cast<clang::ReferenceType>(paramTy)) {
+  } else if (auto refType = dyn_cast<clang::ReferenceType>(paramTy)) {
```
Comment (ravikandhadai) [inline lib/ClangImporter/ImportType.cpp]: Looks like we introduced some whitespace at the end here.

Comment (hyp) [reply lib/ClangImporter/ImportType.cpp]: that's resolved in updated PR

**Comment by hyp** [comment]
@swift-ci please test

**Review by ravikandhadai — APPROVED** [review]
LGTM. Thanks!

File: test/Interop/Cxx/stdlib/use-std-vector.swift [hunk 2]
```diff
@@ -19,18 +19,17 @@ StdVectorTestSuite.test("init") {
 
 StdVectorTestSuite.test("push back") {
     var v = Vector()
-    var _42: CInt = 42
-    v.push_back(&_42)
+    let _42: CInt = 42
+    v.push_back(_42)
```
Comment (zoecarver) [inline test/Interop/Cxx/stdlib/use-std-vector.swift:23]: can we just inline this literal as we do below?

**Review by zoecarver — COMMENTED** [review]
LGTM, thanks Alex (apparently I can't approve my own PR 😆)

**Comment by hyp** [comment]
@swift-ci please test source compatibility

File: test/Interop/Cxx/stdlib/use-std-vector.swift (hunk 2 above)
Comment (hyp) [reply test/Interop/Cxx/stdlib/use-std-vector.swift:23]: Yeah I can do that in a followup. Also need to disable this for template functions too in a follow up.

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Pull request merged**